plugins/modules/cloud/misc/proxmox_snap.py

```diff
@@ -211,6 +238,26 @@ def main():
 
         except Exception as e:
             module.fail_json(msg="Removing snapshot %s of VM %s failed with exception: %s" % (snapname, vmid, to_native(e)))
+    elif state == 'rollback':
+        try:
+            snap_exist = False
+
+            for i in proxmox.snapshot(vm, vmid).get():
+                if i['name'] == snapname:
+                    snap_exist = True
+                    continue
+
+            if not snap_exist:
+                module.exit_json(changed=False, msg="Snapshot %s does not exist" % snapname)
+            else:
+                if proxmox.snapshot_rollback(vm, vmid, timeout, snapname):
+                    if module.check_mode:
+                        module.exit_json(changed=False, msg="Snapshot %s would be rolled back" % snapname)
+                    else:
+                        module.exit_json(changed=True, msg="Snapshot %s rolled back" % snapname)
```

Comment:
Since `exit_json` stops execution, there's no need to indent everything.

I also changed `changed=True` in both cases. Check mode should return the same `changed` status as regular execution.
```suggestion
            if proxmox.snapshot_rollback(vm, vmid, timeout, snapname):
                if module.check_mode:
                    module.exit_json(changed=True, msg="Snapshot %s would be rolled back" % snapname)
                else:
                    module.exit_json(changed=True, msg="Snapshot %s rolled back" % snapname)
```